addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeBaseHandler.java

```diff
@@ -0,0 +1,206 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.solaredge.config.SolarEdgeConfiguration;
+import org.openhab.binding.solaredge.internal.connector.WebInterface;
+import org.openhab.binding.solaredge.internal.model.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolarEdgeBaseHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Friese - initial contribution
+ */
+@NonNullByDefault
+public abstract class SolarEdgeBaseHandler extends BaseThingHandler implements SolarEdgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SolarEdgeBaseHandler.class);
+
+    /**
+     * Interface object for querying the Solaredge web interface
+     */
+    @Nullable
+    private WebInterface webInterface;
+
+    /**
+     * Job which will do the Solaredge polling
+     */
+    private final SolarEdgeLiveDataPolling liveDataPollingRunnable;
+
+    /**
+     * Job which will do the Solaredge polling
+     */
+    private final SolarEdgeAggregateDataPolling aggregateDataPollingRunnable;
+
+    /**
+     * Schedule for polling
+     */
+    @Nullable
+    private ScheduledFuture<?> liveDataPollingJob;
+
+    /**
+     * Schedule for polling
+     */
+    @Nullable
+    private ScheduledFuture<?> aggregateDataPollingJob;
+
+    public SolarEdgeBaseHandler(Thing thing) {
+        super(thing);
+        this.liveDataPollingRunnable = new SolarEdgeLiveDataPolling(this);
+        this.aggregateDataPollingRunnable = new SolarEdgeAggregateDataPolling(this);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("command for {}: {}", channelUID, command);
+        // write access is not supported.
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("About to initialize SolarEdge");
+        Thing thing = this.getThing();
+        SolarEdgeConfiguration config = getConfiguration();
+
+        logger.debug("Solaredge initialized with configuration: {}", config);
+
+        this.webInterface = new WebInterface(config, this);
+
+        if (config.getTokenOrApiKey() != null) {
+            this.startPolling();
+        } else {
+            thing.setStatusInfo(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
```

Comment:
Never set this on the Thing yourself - instead use `updateStatus` methods of the BaseThingHandler.

Reply:
fixed